### DISCUSSION TITLE
using a time column which is not year

### DISCUSSION THREAD

**Discussion by lily-greig** [sdmTMB/sdmTMB#298] [Q&A] [answered]
Hello, the data I am using have a monthly resolution, not yearly. I would like to use the spatiotemporal model. In this case, in which format should my "month+year" time column be ? I tried putting the time column in the format "YYYY-MM" (so as a character). I named this column "year_month", and then used the code:
mod<-sdmTMB(
  x~y+z,
  mesh= mesh,
  data = data,
  time='year_month',
  spatiotemporal = "IID",             
  spatial="on"). 

This seemed to run ok, but when using the 'predict' function (in this case I repredicted on the same data), i.e. 
predictions<-predict(mod, newdata = data), I get errors stating I have NAs in the time column when I don't have any NAs in this year_month column, so I assumed I perhaps used the wrong format. 

What format should I use for time when the data is collected more frequently than yearly? Do you have any other advice for the model run in this case? 

Many thanks, Lily

**Comment by ericward-noaa** [comment, marked as answer]
Hi Lily, this may be happening because time is assumed to be something that can be coerced to numeric internally. Could you try creating a new variable `year_month_n` that represents the months in chronological order, and passing that into the `time` argument?

```
df$year_month <- paste0(df$year, "_", sprintf("%02d", df$month))
df$year_month_n <- as.numeric(as.factor(df$year_month))
```

**Reply by lily-greig** [reply]
Hi Eric, yes, this seems to have worked well. Thank you !

**Comment by seananderson** [comment]
For better or worse, the current method for creating the internal time index is with the function

```r
make_year_i <- function(x) {
  x <- as.integer(as.factor(x))
  x - min(x)
}
```

where `x` is the time column vector. I just thought I would post that in case anybody is trying to debug what's happening with the time column in the future.
